usr/src/lib/libzfs/common/libzfs_dataset.c

```diff
@@ -3642,27 +3642,21 @@ zfs_promote(zfs_handle_t *zhp)
 		return (zfs_error(hdl, EZFS_BADTYPE, errbuf));
 	}
 
-	(void) strlcpy(parent, zhp->zfs_dmustats.dds_origin, sizeof (parent));
-	if (parent[0] == '\0') {
+	if (zhp->zfs_dmustats.dds_origin[0] == '\0') {
 		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 		    "not a cloned filesystem"));
 		return (zfs_error(hdl, EZFS_BADTYPE, errbuf));
 	}
 
-	(void) strlcpy(zc.zc_value, zhp->zfs_dmustats.dds_origin,
-	    sizeof (zc.zc_value));
-	(void) strlcpy(zc.zc_name, zhp->zfs_name, sizeof (zc.zc_name));
-	ret = zfs_ioctl(hdl, ZFS_IOC_PROMOTE, &zc);
+	ret = lzc_promote(zhp->zfs_name, snapname, sizeof(snapname));
 
 	if (ret != 0) {
-		int save_errno = errno;
```

Comment:
the build is failing (see [here](http://jenkins.open-zfs.org/blue/rest/organizations/jenkins/pipelines/openzfs/openzfs/branches/PR-389/runs/4/nodes/97/log/?start=0)) because we've removed `save_errno`, but we're still using it in the call to `zfs_standard_error` below. should we now be passing in `ret` instead of `save_errno` to `zfs_standard_error`?